craft_application/application.py

```diff
@@ -337,21 +337,6 @@ def _configure_services(self, provider_name: str | None) -> None:
             session_policy=self._fetch_service_policy,
         )
 
-    def _resolve_project_path(self, project_dir: pathlib.Path | None) -> pathlib.Path:
```

Comment:
just to double-check: is [snapcraft](https://github.com/canonical/snapcraft/blob/main/snapcraft/application.py#L155) expected to just drop the `override` on their version? Or are there more changes?

Reply:
My concern is that Snapcraft calls that one [extremely early](https://github.com/canonical/snapcraft/blob/main/snapcraft/application.py#L105), before even `_configure_early_services()` has been called

Reply:
I don't think this is worth sweating too much just now; since this is going to a feature branch we'll have time to assess the changes in Snapcraft before setting anything in stone.